### PR TITLE
add a JOSS manuscript in doc/paper

### PR DESCRIPTION
**Pull request by grlee77**
Hi @PyWavelets/core. This PR contains the first revision of a paper for submission to JOSS. Any suggested revisions to the text are welcome. 

The `paper.md` & `paper.bib`need to match the JOSS submission format documented here:
https://joss.readthedocs.io/en/latest/submitting.html

@holgern, as a substantial contributor, I tried to email you about participating in this publication, but did not hear back. If you would like to participate, please send me your affiliation info (and ORCID ID if you have one).

### CODE REVIEW

File: doc/paper/paper.bib [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                  Ankit Agrawal and
+                  SylvainLan and
+                  Michel Pelletier and
+                  Matthew Brett and
+                  Frank Yu and
+                  Daniel M. Pelt and
+                  Saket Choudhary and
+                  Daniele Tricoli and
+                  Check your git settings! and
+                  asnt and
```
Comment (rgommers) [inline doc/paper/paper.bib]: some weirdness in these two lines, and also `0-tree` as the last author. I guess that's how Zenodo does it, but I think we should remove entries that aren't full names. or at least ping them and add them to the mailmap

File: doc/paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+date: 24 August 2018
+bibliography: paper.bib
+---
+
+# Summary
+
+Wavelets are a popular tool for computational harmonic analysis. They provide
+localization in both the temporal (or spatial) domain as well as in the
+frequency domain. A prominant feature is the ability to perform a
```
Comment (rgommers) [inline doc/paper/paper.md]: typo `prominent`

File: doc/paper/paper.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+multiresolution analysis. The wavelet transforms of natural signals and images
+tends to have most of their energy concentrated in a small fraction of the
+coefficients. This sparse representation property is key to the good
+performance of wavelets in applications such as data compression and denoising.
+For example, the wavelet transform is a key component of the JPEG 2000 image
+compression standard.
+
+``PyWavelets`` is a Python package implementing a number of n-dimensinoal
```
Comment (rgommers) [inline doc/paper/paper.md]: typo in `n-dimensional`

File: doc/paper/paper.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+discrete wavelet transforms as well as the 1D continuous wavelet transform. A
+wide variety of predifined wavelets are provided and it is possible for users
```
Comment (rgommers) [inline doc/paper/paper.md]: typo `predifined`.

File: doc/paper/paper.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+to specify custom wavelet filterbanks. All discrete wavelet transforms are
```
Comment (rgommers) [inline doc/paper/paper.md]: "filter bank" should be two words

File: doc/paper/paper.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+``PyWavelets`` was designed for use by scientists working within a range of
+applications including time-series analysis, signal processing, image
+processing and medical imaging. It has already been adopted as a required
+or optional dependency by a number of other software projects. For example,
+it has enabled wavelet-based image denoising in scikit-image [@scikit-image].
+The Operator Discretization Library (ODL) [@odl] uses PyWavelets to enable
+wavelet-based regularization in iterative inverse problems such as computed
+tomography image reconstruction. A number of common 1D demo signals used in
+the literature and in the manuscript by Stephan Mallat [@mallat_book] are
+provided for use in teaching and for purposes of reproducable research. The
```
Comment (rgommers) [inline doc/paper/paper.md]: typo `reproducible`

File: doc/paper/paper.bib (hunk 1 above)
Comment (grlee77) [reply doc/paper/paper.bib]: Hmm. That was exported from Zenodo. I'm not sure what is causing the "Check your git settings!" text to appear there, but will remove it.

Our utils/authors.py maps `asnt` to Alexandre Saint, but does not have an entry for `0-tree` or `jakirkham`. I am pretty sure it is John Kirkham, but don't know for 0-tree. I can ping them and ask. I assume if the user's public GitHub profile doesn't list their name then Zenodo doesn't know how to determine it.

Comment (grlee77) [reply doc/paper/paper.bib]: Do you happen to know if JOSS typically uses these full contributor lists in the references?  In digital format, it doesn't really cost anything to list everyone, so I had just left everyone in. Alternatively, we could truncate after the first six or so (There are six with >= 10 commits, and 17 with > 1 commit) and list a generic "PyWavelets contributors" for the rest if that is preferable. 

The same issue with GitHub usernames shows up in the ODL reference too

Comment (rgommers) [reply doc/paper/paper.bib]: > Do you happen to know if JOSS typically uses these full contributor lists in the references? 

I didn't know, but a quick browse suggests that only the first three authors are listed and then it say something like "n.d."

**Comment by rgommers** [comment]
time to merge, and to submit this?

**Comment by aaren** [comment]
Made an orcid: https://orcid.org/0000-0003-1984-2323

Thanks for the work on this :)

**Comment by grlee77** [comment]
> time to merge, and to submit this?

Yes.  I just merged and submitted it.

**Comment by rgommers** [comment]
Awesome, thanks Greg!

**Pull request merged**